packages/smooth_app/lib/data_models/up_to_date_product_provider.dart

```diff
@@ -1,26 +1,62 @@
-import 'package:flutter/foundation.dart';
-import 'package:flutter/widgets.dart';
 import 'package:openfoodfacts/model/Product.dart';
 import 'package:smooth_app/database/local_database.dart';
 
 /// Provider that reflects all the user changes on [Product]s.
-class UpToDateProductProvider with ChangeNotifier {
-  final Map<String, Product> _map = <String, Product>{};
+class UpToDateProductProvider {
+  UpToDateProductProvider(this.localDatabase);
+
+  final LocalDatabase localDatabase;
+
+  // TODO(monsiertanuki): should be more persistent, like in hive.
+  final Map<String, List<Product>> _map = <String, List<Product>>{};
   final Map<String, int> _timestamps = <String, int>{};
 
-  Product? get(final Product product) => _map[product.barcode!];
+  /// Returns the [product] with all the local pending changes on top.
+  Product getLocalUpToDate(final Product product) {
+    final List<Product>? changes = _map[product.barcode!];
+    if (changes == null) {
```

Comment:
To prevent misused calls, I would add an `isEmpty()` just in case

Reply:
That won't change anything: if `changes` is empty the result will be the same.